uxarray/_latlonbound_utilities.py

```diff
@@ -0,0 +1,141 @@
+import numpy as np
+import copy
+
+from .helpers import angle_of_2_vectors, within, _normalize_in_place, _convert_node_xyz_to_lonlat_rad, \
+    _convert_node_lonlat_rad_to_xyz, _pt_within_gcr
+
+
+def insert_pt_in_latlonbox(old_box_rad, new_pt_rad, is_lon_periodic=True):
+    """Helper function to insert a new point into an existing latlon box. This function will take care of the
+    Longitude wrap-around
+
+    Parameters
+    ----------
+    old_box_rad: float list
+        the original lat lon box using the 2D longitude/latitude in radian [[lat_0, lat_1],[lon_0, lon_1]],required
+    new_pt_rad: float list
+        the inserted new lat lon point [lon, lat], required
+
+    Returns
+    ----------
+    float list
+        the updated lat lon box [[lat_0, lat_1],[lon_0, lon_1]]
+
+    Raises
+    ----------
+    ValueError
+        The longitude of the inserted point is out of range
+    """
+
+    # If the box is null (no point inserted yet)
+
+    if old_box_rad[0][0] == old_box_rad[0][1] == 404.0:
+        latlon_box = old_box_rad
+        latlon_box[0] = [new_pt_rad[0], new_pt_rad[0]]
+
+    if old_box_rad[1][0] == old_box_rad[1][1] == 404.0:
+        latlon_box = old_box_rad
+        latlon_box[1] = [new_pt_rad[1], new_pt_rad[1]]
+
+    if old_box_rad[0][0] == old_box_rad[0][1] == old_box_rad[1][0] == old_box_rad[1][1] == 404.0:
+        return latlon_box
+
+    # Deal with the pole point
+    if new_pt_rad[1] == 404.0 and (
+            (np.absolute(new_pt_rad[0] - 0.5 * np.pi) < 1.0e-12) or (
+            np.absolute(new_pt_rad[0] - (-0.5 * np.pi)) < 1.0e-12)):
+        latlon_box = old_box_rad
+        if np.absolute(new_pt_rad[0] - 0.5 * np.pi) < 1.0e-12:
+            latlon_box[0][1] = 0.5 * np.pi
+        elif np.absolute(new_pt_rad[0] - (-0.5 * np.pi)) < 1.0e-12:
+            latlon_box[0][0] = -0.5 * np.pi
+        return latlon_box
```

Comment:
Also comment from below, can use np.abs() instead of np.absolute() for better readability

Reply:
Machine epsilon is the upper bound on the relative approximation error due to rounding in floating point arithmetic. Therefore, we can only achieve the magnitude lower than that in the calculation

Reply:
> Instead of using an arbitrarily-selected epsilon, I suggest using an order of magnitude higher than machine epsilon defined by one of the following:
> 
> ```python
> np.finfo(float).eps
> np.finfo(np.float32).eps
> np.finfo(np.float64).eps
> ```

Machine epsilon is the upper bound on the relative approximation error due to rounding in floating point arithmetic. Therefore, we can only achieve a magnitude lower than that in the calculation

Reply:
Understood, I was suggesting using a known multiple of the machine epsilon, not comparing directly with the machine epsilon. There's a good read about it [here](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/#:~:text=If%20you%20are%20comparing%20against%20zero%2C%20then%20relative%20epsilons%20and%20ULPs%20based%20comparisons%20are%20usually%20meaningless.%20You%E2%80%99ll%20need%20to%20use%20an%20absolute%20epsilon%2C%20whose%20value%20might%20be%20some%20small%20multiple%20of%20FLT_EPSILON%20and%20the%20inputs%20to%20your%20calculation.%20Maybe). 

I also suggest setting this value as a constant.

Reply:
> Understood, I was suggesting using a known multiple of the machine epsilon, not comparing directly with the machine epsilon. There's a good read about it [here](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/#:~:text=If%20you%20are%20comparing%20against%20zero%2C%20then%20relative%20epsilons%20and%20ULPs%20based%20comparisons%20are%20usually%20meaningless.%20You%E2%80%99ll%20need%20to%20use%20an%20absolute%20epsilon%2C%20whose%20value%20might%20be%20some%20small%20multiple%20of%20FLT_EPSILON%20and%20the%20inputs%20to%20your%20calculation.%20Maybe).
> 
> I also suggest setting this value as a constant.

Got it. The epsilon should be a constant through the entire UXARRAY program indeed. Just we still haven't decided on the exact epsilon at this moment. 

This is still a draft implementation for the algorithm and still requires lots of developer jobs.

Reply:
Of course! This implementation is looking great so far (reviewed in draft since I was requested), but once it's ready for another review feel free to re-request.

Reply:
> Of course! This implementation is looking great so far (reviewed in draft since I was requested), but once it's ready for another review feel free to re-request.

Ohh, I put you as the reviewer because of the antimeridian function as I mentioned before. And thank you again for your valuable advice!